corehq/apps/app_manager/models.py

```diff
@@ -4803,17 +4803,19 @@ def set_media_versions(self, version_reverted_to=None):
         else:
             previous_version = self.get_previous_version()
         prev_multimedia_map = previous_version.multimedia_map if previous_version else {}
-
+        is_linked_app = isinstance(self, LinkedApplication)
         for path, map_item in six.iteritems(self.multimedia_map):
             prev_map_item = prev_multimedia_map.get(path, None)
             if prev_map_item and prev_map_item.unique_id:
                 # Re-use the id so CommCare knows it's the same resource
                 map_item.unique_id = prev_map_item.unique_id
-            if (prev_map_item and prev_map_item.version
-                    and prev_map_item.multimedia_id == map_item.multimedia_id):
-                map_item.version = prev_map_item.version
-            else:
-                map_item.version = self.version
+            # if its a linked app and the map item already has a version don't need to set it up again
```

Comment:
This comment is descriptive but it would help to know *why* this logic exists. Maybe add something like "This prevents versions being unnecessarily updated if a linked app pulls from several substantially different 'streams'"...it's not ideal because the reader needs to understand what streams are, but it's something.

Reply:
Updated, [commit](https://github.com/dimagi/commcare-hq/pull/24285/commits/11647ac8d3963038a8fb358d22e6c76d814feb5b)